File: backend/otgpt_hft/data_model/cmp.py
from typing import Literal

from .abs import DM_Abs, InstanceId

CompareOp = Literal[">", "="]


class DB_ResponseCmp(DM_Abs[Literal["cmp"]]):
    cls: Literal["cmp"] = "cmp"
    # id: InstanceId is prefix with "r_"
    # "a" is better/prefered over "b"
    a: InstanceId
    b: InstanceId
    cmp: CompareOp

File: frontend/lib/data/model/cmp.ts
import { DM_Abs, InstanceId } from "./abs";

export type CMP_T = "cmp";
export const CMP: CMP_T = "cmp";

export type DB_ResponseCmp = DM_Abs<CMP_T> & {
    a: InstanceId
    b: InstanceId
    cmp: ">" | "="
}